src/arrays/programs_involving_multidimensional_data/hard/q1/MainTest.java

```diff
@@ -0,0 +1,164 @@
+package arrays.programs_involving_multidimensional_data.hard.q1;
+
+import global.BaseTest;
+import global.MethodTest;
+import global.exceptions.InvalidClauseException;
+import global.tools.CustomAssertions;
+import global.tools.TestOption;
+import global.utils.ArrayUtil;
+import global.variables.Clause;
+import global.variables.clauses.IntegerLiteral;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.PlaceHolder;
+import global.variables.clauses.StringLiteral;
+import global.variables.wrappers.Optional;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MainTest extends BaseTest {
+    // Java
+
+    public Clause[] testSentence() {
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = "3 2 4 32 4 2 12 32";
+        return new Clause[]{
+                new StringLiteral("Enter the number of rows:"),
```

Comment:
I think instead of asking for input for everything, you should change it to be constant (like 4 by 4), and then we can vary that on the website